contracts/service-registry/src/state.rs

```diff
@@ -145,12 +145,48 @@ pub enum AuthorizationState {
     Jailed,
 }
 
-type ServiceName = str;
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+pub struct VerifierDetailsResponse {
+    pub verifier: Verifier,
+    pub weight: nonempty::Uint128,
+    pub supported_chains: Vec<ChainName>,
+}
+
+#[cw_serde]
+pub struct VerifierChainRecord {
+    pub chain: ChainName,
+    pub verifier: VerifierAddress,
+}
+
+#[index_list(VerifierChainRecord)]
+pub struct VerifierPerChainIndexes<'a> {
+    pub verifier: MultiIndex<
+        'a,
+        VerifierAddress,
+        VerifierChainRecord,
+        (ServiceName, ChainName, VerifierAddress),
+    >,
+}
+
+pub const VERIFIERS_PER_CHAIN_INDEXED_MAP: IndexedMap<
+    (ServiceName, ChainName, VerifierAddress),
+    VerifierChainRecord,
+    VerifierPerChainIndexes,
+> = IndexedMap::new(
+    "verifiers",
+    VerifierPerChainIndexes {
+        verifier: MultiIndex::new(
+            |_pk: &[u8], d| d.verifier.clone(),
+            "verifiers",
+            "verifiers_service_verifier",
```

Comment:
```suggestion
            "verifiers__address",
```

Reply:
keys can have " _ " in them, so a single " _ " between pk namespace and index namespace might not be safe